### DISCUSSION TITLE
Snowflake connector for PowerApps and Power Automate

### DISCUSSION THREAD

**Discussion by VKrishna09** [microsoft/PowerPlatformConnectors#1348] [Ideas]
Hello Microsoft Power Team!

Is there a possibility to get a direct Snowflake connector for PowerApps and Power Automate? 

Power Bi already has one from 5 years! It's time to get this feature as snowflake usage is growing fast these days! 

Anyone would like to add more or have a discussion on this?

**Comment by troystaylor** [comment]
Would this be the API you are requesting?:
https://docs.snowflake.com/en/developer-guide/sql-api/reference.html

**Reply by VKrishna09** [reply]
Hey! Thanks for replying. 

Do you mean to say, we create virtual SQL DB in Snowflake and connect it via SQL server in PowerApps/Automate?

**Reply by troystaylor** [reply]
No, I’m looking for clarification on if there is a REST API that a connector could be built for, and this seems to be the only API I can find. Do you have specific actions that you’d like to use with Power Platform?

**Reply by VKrishna09** [reply]
Snowpipe REST API can be built AFAIK! I have lots of data in Snowflake that needs to be used in the PowerApps(forms) we have.

**Reply by troystaylor** [reply]
So this API?: 
https://docs.snowflake.com/en/user-guide/data-load-snowpipe-rest-apis.html

**Reply by VKrishna09** [reply]
Yes!

**Reply by troystaylor** [reply]
You can give the following YAML a try with a custom connector. You will need to update the host name in the General tab for your host.
https://github.com/troystaylor/Testing/blob/aa56cd32adfaa69261fa8026527fa21cc839f9e4/Snowpipe.yaml
Like @Laskewitz mentioned, it is unlikely that this would be added as an Independent Publisher connector unless the preview connector isn't certified.

**Comment by Laskewitz** [comment]
There already is a Snowflake connector in preview: https://github.com/microsoft/PowerPlatformConnectors/tree/dev/custom-connectors/Snowflake

**Reply by VKrishna09** [reply]
Thats cool! Thanks for sharing! Will it be available as the other connectors are available in Power Platform in near future or do we have to follow this long manual process?

**Reply by Laskewitz** [reply]
For now, it's only available as a manual process. But I think it will be available in-product later. Maybe @natalie-pienkowska can expand on that!

**Reply by Erik-Douma** [reply]
Nice piece of work this custom connector.
Does anyone already have this working within PowerApps? Don't get the JSON data properly parsed in PowerApps. See only [object Object],[object Object],[object Object],[object Object],etc etc after a succesfull request.

**Comment by ausi-kh** [comment]
Any one has an example of using the custom connector for snowflake?
I am able to add the connector and call the executeSQL function, but cannot bind it to the view property of the powerapp objects
